util/patricia_trie/src/nibbleslice.rs

```diff
@@ -154,6 +156,7 @@ impl<'a> NibbleSlice<'a> {
 	}
 
 	/// Encode while nibble slice in prefixed hex notation, noting whether it `is_leaf`.
+	#[inline(always)]
```

Comment:
This function is non-trivial and so I'd probably annotate it `#[inline]` so LLVM can make the choice to inline or not itself. `#[inline(always)]` is only for really trivial functions that are really hot but you have profiled to see are not being inlined automatically (like `len` below) and should be used with care. Although if you've benchmarked this and found that the `(always)` is necessary then ignore this comment, of course!

Reply:
It did show up in the traces and looked suspicious, but now that I try again it doesn't seem to make any difference at all so dialing back to just `#[inline]` makes sense. :)